content/en/blog/_posts/2020-09-03-warnings/index.md

```diff
@@ -0,0 +1,272 @@
+
+```sh
+kubectl get --raw /metrics | prom2json | jq '
+  # set $deprecated to a list of deprecated APIs
+  [
+    .[] | 
+    select(.name=="apiserver_requested_deprecated_apis").metrics[].labels |
+    {group,version,resource}
+  ] as $deprecated 
+  
+  |
+  
+  # select apiserver_request_total metrics which are deprecated
+  .[] | select(.name=="apiserver_request_total").metrics[] |
+  select(.labels | {group,version,resource} as $key | $deprecated | index($key))
+'
+```
+
+Output:
+
+```json
+{
+  "labels": {
+    "code": "0",
+    "component": "apiserver",
+    "contentType": "application/vnd.kubernetes.protobuf;stream=watch",
+    "dry_run": "",
+    "group": "extensions",
+    "resource": "ingresses",
+    "scope": "cluster",
+    "subresource": "",
+    "verb": "WATCH",
+    "version": "v1beta1"
+  },
+  "value": "21"
+}
+{
+  "labels": {
+    "code": "200",
+    "component": "apiserver",
+    "contentType": "application/vnd.kubernetes.protobuf",
+    "dry_run": "",
+    "group": "extensions",
+    "resource": "ingresses",
+    "scope": "cluster",
+    "subresource": "",
+    "verb": "LIST",
+    "version": "v1beta1"
+  },
+  "value": "1"
+}
+{
+  "labels": {
+    "code": "200",
+    "component": "apiserver",
+    "contentType": "application/json",
+    "dry_run": "",
+    "group": "rbac.authorization.k8s.io",
+    "resource": "clusterroles",
+    "scope": "cluster",
+    "subresource": "",
+    "verb": "LIST",
+    "version": "v1beta1"
+  },
+  "value": "1"
+}
+```
+
+The output shows that only read requests are being made to these APIs, and the most requests have been made to watch the deprecated Ingress API.
+
+This information could also be obtained via a Prometheus query.
+The following Prometheus query returns information about requests made to deprecated APIs which will be removed in v1.22:
+
+```promql
+apiserver_requested_deprecated_apis{removed_version="1.22"} * on(group,version,resource,subresource)
+group_right() apiserver_request_total
+```
+
+## Audit annotations
+
+Metrics are a fast way to check whether deprecated APIs are being used, and at what rate,
+but they don't include enough information to identify particular clients or API objects.
+Starting in Kubernetes v1.19, [audit events](https://kubernetes.io/docs/tasks/debug-application-cluster/audit/)
+for requests to deprecated APIs include an audit annotation of `"k8s.io/deprecated":"true"`.
+Administrators can use those audit events to identify specific clients or objects that need to be updated.
+
+# Custom Resource Definitions
+
+Along with the API server ability to warn about deprecated API use, starting in v1.19, a CustomResourceDefinition can indicate a 
+[particular version of the resource it defines is deprecated](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/#version-deprecation).
```

Comment:
same here can do it direct as /docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/#version-deprecation